### ISSUE TITLE
Error when deploying my (first) shinyapp

### ISSUE THREAD

**Issue by alainaa** [rstudio/rmarkdown#924]
Good morning,

I've been recommended (by rstudio/shinyapps) to open an issue here, to see if we're able to fix this issue with my app. In Rstudio they tell me that the app is working fine for them. But not for me...

The app  is  https://alainaa.shinyapps.io/AliOS2017/. Deploying seems to go well, but when the app opens it just shows the error: **Error in if: valor ausente donde TRUE/FALSE es necesario** (something like "error in IF: TRUE/FALSE value expected").

The app runs locally without any problem. These are the session info + the logs, hopefully it helps to understand and solve the issue...

> sessionInfo()

R version 3.3.2 (2016-10-31)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 7 x64 (build 7601) Service Pack 1
locale:
[1] LC_COLLATE=Spanish_Spain.1252  LC_CTYPE=Spanish_Spain.1252    LC_MONETARY=Spanish_Spain.1252
[4] LC_NUMERIC=C                   LC_TIME=Spanish_Spain.1252    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

loaded via a namespace (and not attached):
[1] Rcpp_0.12.8       plotly_4.5.6      knitr_1.15.1      magrittr_1.5      munsell_0.4.3     colorspace_1.3-2 
 [7] viridisLite_0.1.3 R6_2.2.0          stringr_1.1.0     httr_1.2.1        plyr_1.8.4        dplyr_0.5.0      
[13] tools_3.3.2       grid_3.3.2        gtable_0.2.0      DBI_0.5-1         htmltools_0.3.5   yaml_2.1.14      
[19] lazyeval_0.2.0    assertthat_0.1    digest_0.6.10     rprojroot_1.1     tibble_1.2        RJSONIO_1.3-0    
[25] purrr_0.2.2       ggplot2_2.2.0     tidyr_0.6.0       bitops_1.0-6      base64enc_0.1-3   htmlwidgets_0.8  
[31] RCurl_1.95-4.8    rsconnect_0.6     evaluate_0.10     rmarkdown_1.2     stringi_1.1.2     scales_0.4.1     
[37] backports_1.0.4   jsonlite_1.1     
  

And these are the logs:

 

2016-12-20T07:51:10.411183+00:00 shinyapps[146424]: Server version: 1.1.0-116
2016-12-20T07:51:10.564806+00:00 shinyapps[146424]: 
2016-12-20T07:51:10.411215+00:00 shinyapps[146424]: LANG: es_ES.UTF-8
2016-12-20T07:51:10.564809+00:00 shinyapps[146424]: Starting R with process ID: '18'
2016-12-20T07:51:10.411249+00:00 shinyapps[146424]: R version: 3.3.2
2016-12-20T07:51:10.634206+00:00 shinyapps[146424]: 
2016-12-20T07:51:10.411273+00:00 shinyapps[146424]: shiny version: 0.14.2
2016-12-20T07:51:10.411275+00:00 shinyapps[146424]: rmarkdown version: 1.2
2016-12-20T07:51:10.411288+00:00 shinyapps[146424]: knitr version: 1.15.1
2016-12-20T07:51:10.411289+00:00 shinyapps[146424]: jsonlite version: 1.1
2016-12-20T07:51:10.411289+00:00 shinyapps[146424]: RJSONIO version: NA
2016-12-20T07:51:10.411289+00:00 shinyapps[146424]: htmltools version: 0.3.5
2016-12-20T07:51:10.634210+00:00 shinyapps[146424]: Listening on http://0.0.0.0:57292
2016-12-20T07:51:10.559612+00:00 shinyapps[146424]: Using jsonlite for JSON processing
2016-12-20T07:51:13.569109+00:00 shinyapps[146424]: 
2016-12-20T07:51:13.606909+00:00 shinyapps[146424]: Warning: Error in if: valor ausente donde TRUE/FALSE es necesario
2016-12-20T07:51:13.617075+00:00 shinyapps[146424]: Stack trace (innermost first):
2016-12-20T07:51:13.617077+00:00 shinyapps[146424]:     114: discover_rmd_resources
2016-12-20T07:51:13.617079+00:00 shinyapps[146424]:     112: copy_render_intermediates
2016-12-20T07:51:13.617079+00:00 shinyapps[146424]:     111: output_format$intermediates_generator
2016-12-20T07:51:13.617079+00:00 shinyapps[146424]:     110: <Anonymous>
2016-12-20T07:51:13.617080+00:00 shinyapps[146424]:     109: do.call
2016-12-20T07:51:13.617080+00:00 shinyapps[146424]:     108: contextFunc
2016-12-20T07:51:13.617081+00:00 shinyapps[146424]:     107: .getReactiveEnvironment()$runWith
2016-12-20T07:51:13.617081+00:00 shinyapps[146424]:     106: shiny::maskReactiveContext
2016-12-20T07:51:13.617082+00:00 shinyapps[146424]:     105: <reactive>
2016-12-20T07:51:13.617083+00:00 shinyapps[146424]:      94: doc
2016-12-20T07:51:13.617083+00:00 shinyapps[146424]:      93: shiny::renderUI
2016-12-20T07:51:13.617083+00:00 shinyapps[146424]:      92: func
2016-12-20T07:51:13.617084+00:00 shinyapps[146424]:      91: origRenderFunc
2016-12-20T07:51:13.617084+00:00 shinyapps[146424]:      90: output$__reactivedoc__
2016-12-20T07:51:13.617085+00:00 shinyapps[146424]:      15: <Anonymous>
2016-12-20T07:51:13.617078+00:00 shinyapps[146424]:     113: find_external_resources
2016-12-20T07:51:13.617086+00:00 shinyapps[146424]:      12: fn
2016-12-20T07:51:13.617086+00:00 shinyapps[146424]:      11: doTryCatch
2016-12-20T07:51:13.617086+00:00 shinyapps[146424]:      10: tryCatchOne
2016-12-20T07:51:13.617087+00:00 shinyapps[146424]:       9: tryCatchList
2016-12-20T07:51:13.617085+00:00 shinyapps[146424]:      13: rmarkdown::run
2016-12-20T07:51:13.617087+00:00 shinyapps[146424]:       7: connect$retry
2016-12-20T07:51:13.617088+00:00 shinyapps[146424]:       6: eval
2016-12-20T07:51:13.617088+00:00 shinyapps[146424]:       5: eval
2016-12-20T07:51:13.617088+00:00 shinyapps[146424]:       4: eval
2016-12-20T07:51:13.617088+00:00 shinyapps[146424]:       3: eval
2016-12-20T07:51:13.617085+00:00 shinyapps[146424]:      14: do.call
2016-12-20T07:51:13.617089+00:00 shinyapps[146424]:       2: eval.parent
2016-12-20T07:51:13.617087+00:00 shinyapps[146424]:       8: tryCatch
2016-12-20T07:51:13.617089+00:00 shinyapps[146424]:       1: local

¿Any idea? Thank you very much in advance,
Alain.

**Comment by jmcphers** [comment]
Hi Alain,

This is a bug in rmarkdown. We should have it fixed soon, but in the meantime, you should be able to work around it by saving your R Markdown file in the UTF-8 encoding (it looks like it's currently in latin1). Does that solve your problem?

Jonathan.

**Comment by alainaa** [comment]
OK I could fix the issue after changing the encoding to UTF-8. Problem being that it introduces uncorrect spelling of words with “ó”, or  ”ñ” (very common in Spanish). So it will be interesting to fix the bug in rmarkdown.


  And also, I discovered that actually, there was a bug in my app: in some special case I hadn’t detected, an IF condition could not be evaluated. (so the error message displayed by shinyapps.io  “TRUE/FALSE value expected” was really OK!). So even locally, the app was  displaying a warning message in this case (even if it was working well in all the other cases).


  I fixed the bug in my app, and went back to the default encoding: but it didn't work. So back again to the UTF-8!

  Again… Thank you very much for your help!


  Best regards,

  Alain.

  ginal Message -----
